ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -219,6 +225,10 @@ public OutputNode buildQueryLogicalPlan(
       }
     }
 
+    if (!isScalablePush && analysis.getLimitClause().isPresent()) {
```

Comment:
What's the significance of this exclusion?

Reply:
It's true that we could also do this for scalable push, though then we'd probably want to account for it in the physical plan builder as well.  I think it would just ignore it at the moment if you didn't add support there too, which might be fine if we just want to avoid exclusions like this.

If you were going to add support for scalable push, it would probably make sense to do that in a separate PR.

Reply:
Ok, so reading between the lines a little, if we're planning a scalable push query, we simply won't include the limit node in the logical or physical plan. Or are you saying that limit clauses are actually not allowed for scalable push queries at all?

In any case, I'm fine with splitting stuff off into follow-on PRs, but if the clause is supported in SPQs, then I'm concerned about omitting them from the logical plan.

Reply:
They are allowed (and encouraged!), but it would take more code to fully support this logical node.  That's all I meant.